ocs_ci/deployment/multicluster_deployment.py

```diff
@@ -0,0 +1,108 @@
+import logging
+
+from ocs_ci.deployment.deployment import Deployment
+from ocs_ci.framework import config
+from ocs_ci.ocs.exceptions import ACMClusterDeployException
+from ocs_ci.ocs.ui import acm_ui
+from ocs_ci.ocs.utils import get_non_acm_cluster_config
+from ocs_ci.ocs.acm import acm
+from ocs_ci.ocs import constants
+
+
+logger = logging.getLogger(__name__)
+
+
+class OCPDeployWithACM(Deployment):
+    """
+    When we instantiate this class, the assumption is we already have
+    an OCP cluster with ACM installed and current context is ACM
+
+    """
+
+    def __init__(self):
+        """
+        When we init Deployment class it will have all the
+        ACM cluster's context
+        """
+        super().__init__()
+        self.multicluster_mode = config.MULTICLUSTER.get("multicluster_mode", None)
+        # Used for housekeeping during multiple OCP cluster deployments
+        self.deployment_cluster_list = list()
+        # Whether to start deployment in asynchronous mode or synchronous mode
+        # In async deploy mode, we will have a single wait method waiting for
+        # all the cluster deployments to finish
+        self.deploy_sync_mode = config.MULTICLUSTER.get("deploy_sync_mode", "async")
+        self.ui_driver = None
+
+    def do_deploy_ocp(self, log_cli_level="INFO"):
+        """
+        This function overrides the parent's function in order accomodate
+        ACM based OCP cluster deployments
+
+        """
+        if self.multicluster_mode == constants.RDR_MODE:
+            self.do_rdr_acm_ocp_deploy()
+
+    def do_rdr_acm_ocp_deploy(self):
+        """
+        Specific to regional DR OCP cluster deployments
+
+        """
+        factory = acm_ui.ACMOCPDeploymentFactory()
+        self.ui_driver = acm.login_to_acm()
```

Comment:
Why the object name is ui_driver? This can be really confusing.

Reply:
Because login_to_acm returns a selenium driver right ?

Reply:
yes, but using the `driver` word in the object name could be confusing. Just a thought